File: program/utils/isPalindrome.go
package utils

import (
	"fmt"
)

func IsPalindrome(x int) bool {
	var result = 0
	var t = x
	var digit = 0
	if x < 0 {
		return false
	}

	for t != 0 {
		digit = t % 10
		t /= 10
		result = result*10 + digit
	}

	if result != x {
		fmt.Println("Not palindrome")
		return false
	}
	fmt.Println("palindrome")
	return true
}

File: c_c++/isPalindrome.cpp
#include <stdio.h>
#include <iostream>
#include <vector>

using namespace std;


class Solution{
    public:
    void isPalindrome(int x){
        int t =x;
        int digit=0;

        unsigned int result = 0;
        if(t>0){
            while(t!=0){
            digit = t%10;
            t/=10;
            result = result * 10 + digit;
            }
        }   
        
        if (result == x){
            printf("Palindrome\n");
        }else{
            printf("Not a Palindrome\n");
        }
    }
};

int main(){
    Solution a;
    a.isPalindrome(121);

    return 0;
}